### PR TITLE
Fixes Profile Icon reload disappearing issue

### LINKED ISSUE #590: 🐞 [BUG] - Profile Icon of navbar disappeared on reloading page.
### Description

The profile icon of NavBar is converted into Signup button on page reload, even if the user is logged in. 
Possible reason behind this, we are not updating `auth contex`t on reload.

### Current behaviour

N/A

### Expected behaviour

Navbar should show profile icon even of reload.

### Reproduction steps

```bash
1. Log in to website
2. There will be a profile icon present
3. Refresh page.
```

### Screenshots

_No response_

### Logs

_No response_

### Browsers

Firefox, Chrome

### OS

Windows

### Are you contributing for a program ?

GSSoC2023

### Code of Conduct

- [X] I have read and agree to the Contributing Guidelines of this project.

### PR DESCRIPTION
**Pull request by Isha988**
## Related Issue

Closes #590 

## Description of Changes
Added functionality to update auth state on reload

### CODE REVIEW

File: context/AuthContext.js [hunk 1]
```diff
@@ -10,6 +10,24 @@ export function AuthProvider(props) {
   const [isMentorLoggedIn, setIsMentorLoggedIn] = useState(null);
   const [isUserLoggedIn, setIsUserLoggedIn] = useState(null);
 
+  useEffect(() => {
+    // fetching user from local storage
```
Comment (anmode) [inline context/AuthContext.js:14]: We have to decrypt first...see userprofile.js

**Review by anmode — CHANGES_REQUESTED** [review]
Cool,

File: context/AuthContext.js (hunk 1 above)
Comment (Isha988) [reply context/AuthContext.js:14]: But why, I am not using any value of objects, I am just checking whether userData or mentorData object is there or not

**Comment by anmode** [comment]
Mybad! Yes you are correct

**Review by muhammadjufry — CHANGES_REQUESTED** [review]
Cool!, Btw could you please reduce the shadow for the icon? Because the shadow is too bold

**Comment by Isha988** [comment]
> Cool!, Btw could you please reduce the shadow for the icon? Because the shadow is too bold

Screenshot 2023-07-24 at 11-54-36 GrabTern Grab Your Internship
Profile icon is not having any shadow

**Comment by muhammadjufry** [comment]
I mean in forgot password page

**Comment by Isha988** [comment]
> I mean in forgot password page

okay, wait I am looking it.

**Pull request merged**